src/iam.ts

```diff
@@ -262,14 +263,28 @@ export class IAM extends IAMBase {
   async updateDidDocument(options: {
     didAttribute: DIDAttribute;
     data: IUpdateData;
+    did?: string;
     validity?: number;
   }): Promise<boolean> {
-    const { didAttribute, data, validity } = options;
-    if (this._document) {
+    const { didAttribute, data, validity, did } = options;
+
+    if (!this._document) {
+      throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
+    }
```

Comment:
Looks like you don't need `this._document` if `did` is passed.
Same for `this._didSigner` if `did` is passed

Reply:
I need `this._didSigner` when `did` is passed. Without that i can't create ProxyOperator, or can I?